chapters/wsi.md

```diff
@@ -38,4 +38,4 @@ Khronos' [sample](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/sam
 
 Mobile devices can be rotated, therefore the logical orientation of the application window and the physical orientation of the display may not match. Applications need to be able to operate in two modes: `portrait` and `landscape`. The difference between these two modes can be simplified to just a change in resolution. However, some display subsystems always work on the "native" (or "physical") orientation of the display panel. Since the device has been rotated, to achieve the desired effect the application output must also rotate.
 
-In order for your application to get the most out of Vulkan on mobile platforms, such as Android, implementing pre-rotation is a must. There is a [detailed blog post from Google](https://android-developers.googleblog.com/2020/02/handling-device-orientation-efficiently.html?m=1) that goes over how to handle the surface rotation by specifying the orientation during swapchain creation and also comes with a [standalone example](https://github.com/google/vulkan-pre-rotation-demo). The [Vulkan-Samples](https://github.com/KhronosGroup/Vulkan-Samples) also has both a [great write up](https://github.com/KhronosGroup/Vulkan-Samples/blob/master/samples/performance/surface_rotation/surface_rotation_tutorial.md) of why pre-rotation is a problem as well as [a sample to run](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/samples/performance/surface_rotation) that shows a way to solve it in the shader.
+In order for your application to get the most out of Vulkan on mobile platforms, such as Android, implementing pre-rotation is a must. There is a [detailed blog post from Google](https://android-developers.googleblog.com/2020/02/handling-device-orientation-efficiently.html?m=1) that goes over how to handle the surface rotation by specifying the orientation during swapchain creation and also comes with a [standalone example](https://github.com/google/vulkan-pre-rotation-demo). The [Vulkan-Samples](https://github.com/KhronosGroup/Vulkan-Samples) also has both a [great write up](https://github.com/KhronosGroup/Vulkan-Samples/blob/master/samples/performance/surface_rotation/surface_rotation_tutorial.md) of why pre-rotation is a problem as well as [a sample to run](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/samples/performance/surface_rotation) that shows a way to solve it in the shader. Qualcomm suggest extension [VK_QCOM_render_pass_transform](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_QCOM_render_pass_transform.html) to implement pre-rotation for Adreno devices.
```

Comment:
looks good to me otherwise

```suggestion
In order for your application to get the most out of Vulkan on mobile platforms, such as Android, implementing pre-rotation is a must. There is a [detailed blog post from Google](https://android-developers.googleblog.com/2020/02/handling-device-orientation-efficiently.html?m=1) that goes over how to handle the surface rotation by specifying the orientation during swapchain creation and also comes with a [standalone example](https://github.com/google/vulkan-pre-rotation-demo). The [Vulkan-Samples](https://github.com/KhronosGroup/Vulkan-Samples) also has both a [great write up](https://github.com/KhronosGroup/Vulkan-Samples/blob/master/samples/performance/surface_rotation/surface_rotation_tutorial.md) of why pre-rotation is a problem as well as [a sample to run](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/samples/performance/surface_rotation) that shows a way to solve it in the shader. If using an Adreno GPU powered device, Qualcomm suggests making use of the [VK_QCOM_render_pass_transform](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_QCOM_render_pass_transform.html) extension to implement pre-rotation.
```